pkg/suggestion/v1alpha2/parameter.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from sklearn.preprocessing import MinMaxScaler
+
+class ParameterConfig:
+    """
+    Class to hold the parameter configuration for an experiment.
+
+    Attributes:
+        name_ids (dict): Mapping from a parameter name to the index of that
+            parameter in the other fields.
+        dim (int): Dimension of the vectors created when parameter assignments
+            are mapped to a vector. Each int, double, or discrete parameter
+            adds one to the dimension, and each categorical parameter adds
+            the number of feasible values for that parameter due to one-hot
+            encoding.
+        lower_bounds (ndarray): The lower bounds for each parameter in the
+            search space.
+        upper_bounds (ndarray): The lower bounds for each parameter in the
+            search space.
+        parameter_types (list): The type of each parameter.
+        names (list): The name of each parameter.
+        discrete_info (list): A list of dicts where each dict contains the
+            information for a single discrete parameter. An example of a dict
+            is {"name": "discrete_parameter, "values": [2, 3, 5]}]
+        categorical_info (list): A list of dicts where each dict contains the
+            information for a single categorical parameter. An example dict is
+            {"name": "cat_param", "values": ["true", "false"], "number": 2}.
+    """
+
+    def __init__(self, name_ids, dim, lower_bounds, upper_bounds,
+                 parameter_types, names, discrete_info, categorical_info):
+        self.name_ids = name_ids
+        self.dim = dim
+        self.lower_bounds = np.array(lower_bounds).reshape((1, dim))
+        self.upper_bounds = np.array(upper_bounds).reshape((1, dim))
+        self.parameter_types = parameter_types
+        self.names = names
+        self.discrete_info = discrete_info
+        self.categorical_info = categorical_info
+        if len(self.names) != len(set(self.names)):
+            raise Exception("Parameter names are not unique.")
+
+    def create_scaler(self):
```

Comment:
Do these methods need to be in this class? They seem to be specific to Bayestian optimization and random search respectively, so maybe they should just be in the specific algorithms.

Also the file name should be parameter_config.py.